src/main/scala/server/ServerApp.scala

```diff
@@ -8,40 +8,61 @@ import org.http4s.dsl.io._
 import org.http4s.implicits._
 import io.circe.generic.auto._
 import org.http4s.circe._
-import Domain.{Subscription, SubscriptionRequest}
+import Domain.{Subscription, SubscriptionRequest, PutRequest}
 
 import java.util.UUID
 import scala.concurrent.ExecutionContext.global
 
 object ServerApp extends IOApp {
 
-  implicit val decoder: EntityDecoder[IO, SubscriptionRequest] = jsonOf[IO, SubscriptionRequest]
+  implicit val decoderPost: EntityDecoder[IO, SubscriptionRequest] = jsonOf[IO, SubscriptionRequest]
+  implicit val decoderPut: EntityDecoder[IO, PutRequest] = jsonOf[IO, PutRequest]
 
   override def run(args: List[String]): IO[ExitCode] = {
-    var storage: Map[String, Subscription] = Map.empty
+    implicit var storage: Map[String, Subscription] = Map.empty
+
+    def existsId(id: String, request: Either[String, PutRequest])(implicit storage: Map[String, Subscription]): Either[String, PutRequest] = {
```

Comment:
actually you can refer to `storage` not having it as implicit param. `existsId` is in the scope of `run`, so it has access to `storage` variable.

Reply:
true 👍